app/src/apdu_handler.c

```diff
@@ -73,6 +77,20 @@ __Z_INLINE bool process_chunk(__Z_UNUSED volatile uint32_t *tx, uint32_t rx)
         THROW(APDU_CODE_WRONG_LENGTH);
     }
 
+    uint8_t num_of_txns = P1_NUM_OF_TXNS_IN_GROUP(P1);
+
+    if (num_of_txns != 0) {
+        if (num_of_txns > 16 || num_of_txns == 1) {
```

Comment:
maybe use defined constants for those limits?

making the code easier to read and to adjust later